presto-main/src/main/java/com/facebook/presto/metadata/FunctionRegistry.java

```diff
@@ -198,6 +198,7 @@
 import static com.facebook.presto.operator.aggregation.MinByNAggregationFunction.MIN_BY_N_AGGREGATION;
```

Comment:
I split out a few refactoring and preparatory commits.

I didn't split things out following suggestions from @kokosing. To me, it's more important to make sure that two unrelated changes that mix together doesn't go in the same commit.

`Apply` function is largely unrelated to the rest of the code. It's act as a test mostly. As a result, I don't see how splitting that out can make code review easier. Additionally, without the apply function, code generation and semantic analyzer will basically go untested.

I didn't spend the time to split out code generation and semantic analyzer for the same reason. 1. Additional tests will be needed if I do so. 2. They are not really mixed together anyways.